validator/errorbundler.py

```diff
@@ -116,6 +120,44 @@ def wrap(self, *args, **kwargs):
     warning = _message("warnings", "warning")
     notice = _message("notices", "notice")
 
+    def system_error(self, msg_id=None, message=None, description=None,
+                     validation_timeout=False, exc_info=None):
+        """Add an error message for an unexpected exception in validator
+        code, and move it to the front of the error message list. If
+        `exc_info` is supplied, the error will be logged.
+
+        If the error is a validation timeout, it is re-raised unless
+        `msg_id` is "validation_timeout"."""
+
+        if exc_info:
+            if (isinstance(exc_info[1], validator.ValidationTimeout) and
+                    msg_id != "validation_timeout"):
+                # These should always propagate to the top-level exception
+                # handler, and be reported only once.
```

Comment:
The fact that we re-raise before we actually report it if we're not at the top-level.